test/Microsoft.TestPlatform.AcceptanceTests/CodeCoverageTests.cs

```diff
@@ -0,0 +1,157 @@
+            var arguments = PrepareArguments(assemblyPaths, this.GetTestAdapterPath(), runSettings,
+                this.FrameworkArgValue, runnerInfo.InIsolationValue);
+            arguments = string.Concat(arguments, $" /ResultsDirectory:{resultsDirectory}", $" /Diag:{diagFileName}",
+                $" /TestAdapterPath:{traceDataCollectorDir}", " /Collect:\"Code Coverage\"");
+            arguments = string.Concat(arguments, $" /Platform:{targetPlatform}");
+
+            var trxFilePath = Path.Combine(this.resultsDirectory, Guid.NewGuid() + ".trx");
+            arguments = string.Concat(arguments, " /logger:trx;logfilename=" + trxFilePath);
+
+            this.InvokeVsTest(arguments);
+
+            this.ValidateSummaryStatus(1, 1, 1);
+
+            var actualCoverageFile = CodeCoverageTests.GetCoverageFileNameFromTrx(trxFilePath, resultsDirectory);
+            Console.WriteLine($@"Coverage file: {actualCoverageFile}  Results directory: {resultsDirectory} trxfile: {trxFilePath}");
+            Assert.IsTrue(File.Exists(actualCoverageFile), "Coverage file not found: {0}", actualCoverageFile);
+
+            // Microsoft.VisualStudio.Coverage.Analysis assembly not avaialble for .NET Core.
+#if NET451
+            this.ValidateCoverageData(actualCoverageFile);
+#endif
+            Directory.Delete(this.resultsDirectory, true);
+        }
+
+#if NET451
+        private void ValidateCoverageData(string coverageFile)
+        {
+            using (var converageInfo = CoverageInfo.CreateFromFile(coverageFile))
+            {
+                CoverageDS coverageDs = converageInfo.BuildDataSet();
+                AssertModuleCoverageCollected(coverageDs);
+                AssertSourceFileName(coverageDs);
+            }
+
+        }
+
+        private static void AssertSourceFileName(CoverageDS coverageDS)
+        {
+            var sourceFileNames = from sourceFilePath in coverageDS.GetSourceFiles()
+                select Path.GetFileName(sourceFilePath);
+            var expectedFileName = "UnitTest1.cs";
+            CollectionAssert.Contains(
+                sourceFileNames.ToArray(),
+                expectedFileName,
+                $"Code Coverage not collected for file: {expectedFileName}");
+        }
+
+        private void AssertModuleCoverageCollected(CoverageDS coverageDS)
+        {
+            var moduleFound = false;
+            for (int i = 0; i < coverageDS.Module.Count; i++)
+            {
+                var module = coverageDS.Module[i];
+                if (module.ModuleName.Equals(assemblyName, StringComparison.OrdinalIgnoreCase))
+                {
+                    moduleFound = true;
+                    break;
+                }
+            }
+
+            Assert.IsTrue(moduleFound, $"Code coverage not collected for module: {assemblyName}");
+        }
+#endif
+
+        private static string GetCoverageFileNameFromTrx(string trxFilePath, string resultsDirectory)
```

Comment:
>GetCoverageFileNameFromTrx [](start = 30, length = 26)

Rename this method to include verification part

Reply:
This function is getting called from tests. IMO function name no need to represent how it is achieving the task.